src/OpenTelemetry.Instrumentation.Http/Implementation/HttpHandlerDiagnosticListener.cs

```diff
@@ -300,4 +301,67 @@ public void OnException(Activity activity, object payload)
+#endif
+    private bool TryFetchException(object payload, out Exception exc)
+    {
+        if (!this.stopExceptionFetcher.TryFetch(payload, out exc) || exc == null)
+        {
+            return false;
+        }
+
+        return true;
+    }
+
+    // The AOT-annotation DynamicallyAccessedMembers(https://learn.microsoft.com/dotnet/api/system.diagnostics.codeanalysis.dynamicallyaccessedmembersattribute?view=net-7.0)
+    // in Systm.Net.Http library ensures that top-level properties on the payload object are always preserved.
+    // see https://github.com/dotnet/runtime/blob/f9246538e3d49b90b0e9128d7b1defef57cd6911/src/libraries/System.Net.Http/src/System/Net/Http/DiagnosticsHandler.cs#L325
+#if NET6_0_OR_GREATER
+    [UnconditionalSuppressMessage("Trimming", "IL2026", Justification = "The event source guarantees that top-level properties are preserved")]
+#endif
+    private bool TryFetchRequest(object payload, out HttpRequestMessage request)
+    {
+        if (!this.startRequestFetcher.TryFetch(payload, out request) || request == null)
+        {
+            return false;
+        }
+
+        return true;
+    }
+
+    // The AOT-annotation DynamicallyAccessedMembers(https://learn.microsoft.com/dotnet/api/system.diagnostics.codeanalysis.dynamicallyaccessedmembersattribute?view=net-7.0)
+    // in Systm.Net.Http library ensures that top-level properties on the payload object are always preserved.
+    // see https://github.com/dotnet/runtime/blob/f9246538e3d49b90b0e9128d7b1defef57cd6911/src/libraries/System.Net.Http/src/System/Net/Http/DiagnosticsHandler.cs#L325
+#if NET6_0_OR_GREATER
+    [UnconditionalSuppressMessage("Trimming", "IL2026", Justification = "The event source guarantees that top-level properties are preserved")]
+#endif
+    private bool TryFetchResponse(object payload, out HttpResponseMessage response)
+    {
+        if (this.stopResponseFetcher.TryFetch(payload, out response) && response != null)
+        {
+            return true;
+        }
+
+        return false;
+    }
+
+    // The AOT-annotation DynamicallyAccessedMembers(https://learn.microsoft.com/dotnet/api/system.diagnostics.codeanalysis.dynamicallyaccessedmembersattribute?view=net-7.0)
+    // in Systm.Net.Http library ensures that top-level properties on the payload object are always preserved.
+    // see https://github.com/dotnet/runtime/blob/f9246538e3d49b90b0e9128d7b1defef57cd6911/src/libraries/System.Net.Http/src/System/Net/Http/DiagnosticsHandler.cs#L325
+#if NET6_0_OR_GREATER
+    [UnconditionalSuppressMessage("Trimming", "IL2026", Justification = "The event source guarantees that top-level properties are preserved")]
+#endif
+    private TaskStatus GetRequestStatus(object payload)
+    {
+        //// https://github.com/dotnet/runtime/blob/master/src/libraries/System.Net.Http/src/System/Net/Http/DiagnosticsHandler.cs
+        //// requestTaskStatus is not null
```

Comment:
What exactly does `requestTaskStatus is not null` mean? It is a non-nullable enum property.

Reply:
https://github.com/open-telemetry/opentelemetry-dotnet/pull/4770#discussion_r1295236878